compiler-core/src/type_/error.rs

```diff
@@ -789,6 +791,20 @@ pub enum UnknownValueConstructorError {
     },
 }
 
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+pub enum UnknownValueConstructorErrorSituation {
+    /// A value using ```let``` is assigned to a variable having a capital case.
+    UnknownConstructorName,
```

Comment:
This name doesn't match the description or the error produced here. Something to do with patterns perhaps?